### DISCUSSION TITLE
[FR][EN] Facture électronique/Electronic invoice

### DISCUSSION THREAD

**Discussion by valodiadeseynes** [baimard/gestion#201] [Ideas]
Bonjour,

Sera-t-il possible de générer des factures au format électronique ? Il existe une norme européenne : le format [Factur-X](https://fnfe-mpe.org/factur-x/) qui est une facture PDF classique contenant un fichier XML permettant le traitement comptable automatisé du document.

Etant donné que dès 2025 toute société française devra être capable de générer des factures électroniques, je pense que c'est une fonctionnalité intéressante à apporter.

**Comment by baimard** [comment]
Bonjour,

effectivement, je vais me renseigner sur cette possibilité, mais si c'est réellement un standard je vais l'implémenter !

Merci beaucoup de votre retour.

**Reply by Greg7527** [reply]
Bonjour,

C'est un format prévu
https://factur-x.org/
(Voir aussi le site de la FNFE)

L'obligation d'émission s'impose à tous au 1er septembre 2027, mais l'obligation de réception est au 1er septembre 2026
https://www.impots.gouv.fr/professionnel/je-decouvre-la-facturation-electronique

Franchement, bon courage !
https://www.impots.gouv.fr/fiches-pedagogiques-pour-mieux-comprendre-la-facturation-electronique

Une API vers une PDP serait peut-être plus simple ?

**Reply by Greg7527** [reply]
Une solution est possible : le transfert de la facture sur jefacture.com
https://ecma-solutions.com/produits-entreprise/jefacture

Ou toute autre solution équivalente. (ce qui règlerait le problème français, mais pas européen => PEPOL)

**Comment by baimard** [comment]
Normalement la génération d'un format XML devrait être possible très facilement, mais je ne trouve pas la référence documentaire sur le format du XML.

après n'oublions pas que c'est pour le BtoB.

En suite, je pense qu'un simple ajout d'un bouton générer Facture électronique sera mis à côté de la génération PDF. et ça fera un fichier XML.

**Reply by Greg7527** [reply]
J'ai vu cela quelque part...
Ici ?
https://fnfe-mpe.org/factur-x/implementer-factur-x/

**Comment by baimard** [comment]
Bonjour à tous,
J'ai lu la réglementation sur la facture électronique. Malheureusement, les formats de collaboration électronique ne sont pas encore officiels. Tout ce qui se trouve sur Internet est soit très partiel, soit non officiel, notamment des tentatives de certaines personnes d'imposer leurs référentiels.

Dès que la réglementation européenne établissant le format sera publiée, nous la partagerons, et je procéderai à l'intégration de la fonctionnalité. Cela devrait être assez rapide.

-----------
Hello everyone,
I have read the regulations on electronic invoicing. Unfortunately, the electronic collaboration formats are not yet official. Everything available online is either very incomplete or unofficial, notably some individuals trying to impose their own standards.

As soon as the European regulation defining the format is published, we will share it, and I will proceed with integrating the functionality. This should be quite quick.

**Comment by baimard** [comment]
### 1. Contexte de la réglementation sur la facture électronique

- Directive européenne : La directive européenne e-Invoicing (2014/55/EU) vise à harmoniser les factures électroniques dans l'Union européenne, notamment dans le cadre des marchés publics. Cependant, son application dans les transactions B2B/B2C reste à préciser.
- Normes existantes : EN 16931 est la norme européenne pour les factures électroniques, mais elle est principalement orientée vers le secteur public. Elle pourrait influencer les formats pour d'autres usages.

### 2. Formats et protocoles courants

- Factures structurées : Formats comme UBL (Universal Business Language) et Factur-X (un format hybride utilisé en France, combinant XML structuré et PDF lisible).
- PEPPOL : Un réseau et un standard d'échange électronique largement utilisé pour les factures dans plusieurs pays européens.
- Beaucoup de discussions et d'expérimentations sont encore en cours pour les rendre compatibles avec les futurs standards européens obligatoires.

### 3. Les défis actuels

- Fragmentation : Chaque pays adopte des approches légèrement différentes (ex. : Chorus Pro en France, SDI en Italie). Ces systèmes nationaux pourraient influencer les standards européens.
- Manque d'uniformité : Comme vous l'avez mentionné, les informations disponibles sont souvent incomplètes ou non officielles. Certaines initiatives privées ou nationales peuvent prêter à confusion.

### 4. Prochaines étapes

- La Commission européenne travaille actuellement sur des standards pour uniformiser les formats et l'échange électronique dans l'Union européenne. Ces standards devraient clarifier les choses d'ici quelques mois ou années.
- Une fois la réglementation publiée, il faudra s'assurer que votre application est compatible avec les standards définis, que ce soit via des modules natifs ou des API pour intégrer des solutions comme PEPPOL.




--------------------------------


### 1. Context of Electronic Invoicing Regulation

- **European Directive**: The European e-Invoicing Directive (2014/55/EU) aims to harmonize electronic invoices within the European Union, particularly in the context of public procurement. However, its application to B2B/B2C transactions remains to be clarified.
- **Existing Standards**: EN 16931 is the European standard for electronic invoices, primarily focused on the public sector. It may influence formats for other use cases.

### 2. Common Formats and Protocols

- **Structured Invoices**: Formats such as UBL (Universal Business Language) and Factur-X (a hybrid format used in France that combines structured XML and readable PDF).
- **PEPPOL**: A network and standard for electronic exchange widely used for invoicing in several European countries.
- Significant discussions and experiments are ongoing to ensure compatibility with future mandatory European standards.

### 3. Current Challenges

- **Fragmentation**: Each country adopts slightly different approaches (e.g., Chorus Pro in France, SDI in Italy). These national systems could shape European standards.
- **Lack of Uniformity**: As mentioned, available information is often incomplete or unofficial. Certain private or national initiatives can create confusion.

### 4. Next Steps

- The **European Commission** is currently working on standards to unify formats and electronic exchanges across the European Union. These standards should bring clarity in the coming months or years.
- Once the regulation is published, it will be essential to ensure that your application is compatible with the defined standards, either through native modules or APIs to integrate solutions like PEPPOL.

---------------------------------------------------------

**Comment by baimard** [comment]
https://www.impots.gouv.fr/sites/default/files/media/1_metier/2_professionnel/EV/2_gestion/290_facturation_electronique/specification_externes_b2b/specifications-externes-v3.0.zip

Je viens de voir la liste des spécifications pour devenir PDP, il faut être ISO 27001, et avoir un serveur SECNUMCLOUD ... principalement. Le reste c'est de la documentation ...

Je travaille dessus mais je pense que je vais devoir ouvrir un consortium ... Il n'y a que des plateformes payantes actuellement ...

**Comment by solarpush** [comment]
Bonjour,

Je tombe sur cette discussion et je me permets de réagir : le format Facture-X est bel et bien officiel et normé depuis plusieurs années déjà (EN 16931, profil Facture-X 1.08 publié par la FNFE-MPE). La spec du XML CII (Cross Industry Invoice D22B) est documentée et validée — le validateur FNFE est dispo publiquement.

Par contre je confirme que devenir PDP c'est une autre paire de manches (ISO 27001, SecNumCloud…), mais générer une facture au format Facture-X n'a rien à voir avec le statut PDP. Une facture électronique Facture-X c'est juste un PDF/A-3 avec un XML CII EN16931 embarqué. Pas besoin d'être PDP pour ça.

On a justement développé un outil open source qui fait exactement ça : JSON en entrée → PDF Facture-X valide EN16931 + EXTENDED en sortie. Go + Angular + Typst, ~165 factures/seconde, sans headless browser.

Ça peut s'intégrer comme microservice externe à baimard/gestion sans une ligne à toucher côté PHP : le bouton « Facture-X » appelle l'API, le PDF est streamé en retour.

Repo : https://github.com/solarpush/fX
Article technique (FR) : https://medium.com/@pierre.mjenr/comment-nous-avons-réinventé-la-génération-de-factures-avec-go-angular-et-typst-pour-facturx-7a749f53a8c0

N'hésitez pas si vous voulez qu'on en parle, c'est en AGPL donc compatible avec votre projet.

**Reply by baimard** [reply]
Trop on a déjà codé le module, c'était il y a un mois :'(

**Reply by solarpush** [reply]
Aucun souci, nous continuerons de faire évoluer pour toucher aussi la gen haute performance de pdf hors facture et facturX via micro service.
Ça sera pour une prochaine bonne soirée.

**Comment by baimard** [comment]
Bonjour à tous 👋,

Je suis sur le point de publier la nouvelle version intégrant la **facturation électronique** ! 🎉

Quand je pense que nous avions identifié ce besoin il y a déjà **5 ans**… Ce fut un important travail d'évolution, de refonte et de maturation de l'application pour en arriver là.

Cette première étape permet désormais de générer des factures électroniques au format **Factur-X**. Mais ce n'est que le début !

Le prochain grand chantier sera de connecter Gestion au plus grand nombre possible de plateformes de facturation électronique afin de simplifier encore davantage vos échanges.

Merci à toutes les personnes qui suivent le projet et contribuent à son évolution, nous sommes 3000 à utiliser l'application. L'aventure continue ! 🚀

**Reply by valodiadeseynes** [reply]
Pour les PDP (enfin maintenant on appelle ça des PA ou Plateformes Agréées) je te conseille de garder un oeil sur le projet PDP Libre qui vise à créer une plateforme communautaire à long terme.
Sinon, la PA qui semble être la plus ouverte niveau intégration c'est SuperPDP. Je sais que Jefacture propose aussi une API mais il est un peu plus difficile d'y avoir accès.

**Reply by baimard** [reply]
Tu crois qu'il va y avoir des plateformes communautaires ? ça serait extra pour ce projet !

On va finir par avoir un vrai logiciel de compta complet ah ah

**Comment by baimard** [comment]
Bonjour, si vous avez une plateforme préférée c'est le moment de me le dire ! 

je suis entrain de partir sur weinvoice ...
